### PR TITLE
Increase coverage of python-gvm

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

* Increase coverage of `connections.py`
* Increase coverage of `xml.py`
* Remove AutoAddPolicy() from paramiko to avoid MITM-attack potential [source](https://github.com/github/codeql/blob/main/python/ql/src/Security/CWE-295/MissingHostKeyValidation.ql) [#481](https://github.com/greenbone/python-gvm/pull/481)

**Why**:

**How**:

**Checklist**:

- [x] Tests
- [ ] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [ ] Documentation

### CODE REVIEW

File: tests/connections/test_ssh_connection.py [hunk 1]
```diff
@@ -46,3 +48,111 @@ def check_ssh_connection_for_default_values(self, ssh_connection):
...
+    def test_send(self):
+        with patch('paramiko.SSHClient') as SSHClientMock:
+            client_mock = SSHClientMock.return_value
+            stdin = Mock()
+            stdin.channel.send.return_value = 4
+            client_mock.exec_command.return_value = [stdin, None, None]
+            ssh_connection = SSHConnection()
+
+            ssh_connection.connect()
+            ssh_connection.send("blah")
```
Comment (bjoernricks) [inline tests/connections/test_ssh_connection.py:118]: I am missing some testing of the outcome here. It's just running code but not ensuring some behavior.

File: tests/connections/test_unix_socket_connection.py [hunk 2]
```diff
@@ -47,9 +48,9 @@ class ThreadedUnixStreamServer(
 
 
 class UnixSocketConnectionTestCase(unittest.TestCase):
-    # pylint: disable=protected-access
+    # pylint: disable=protected-access, invalid-name
     def setUp(self):
-        self.socketname = "%s/%s.sock" % (
+        self.socketname = "{}/{}.sock".format(
```
Comment (bjoernricks) [inline tests/connections/test_unix_socket_connection.py:53]: I think we should use f-strings now for changes string formatting.

File: tests/connections/test_unix_socket_connection.py [hunk 3]
```diff
@@ -68,37 +69,60 @@ def tearDown(self):
...
+        connection.connect()
+        connection.send("<gmp/>")
+        connection.read()
+        connection.disconnect()
+
+    def test_unix_socket_connect_file_not_found(self):
+        connection = UnixSocketConnection(path="foo", timeout=DEFAULT_TIMEOUT)
+        with self.assertRaises(GvmError) as e:
+            connection.connect()
+            self.assertEqual(str(e), 'Socket foo does not exist')
```
Comment (bjoernricks) [inline tests/connections/test_unix_socket_connection.py:101]: assertRaises can get a second parameter for testing the error message directly. see https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

File: tests/connections/test_unix_socket_connection.py [hunk 4]
```diff
@@ -68,37 +69,60 @@ def tearDown(self):
...
             path=self.socketname, timeout=DEFAULT_TIMEOUT
         )
-        self.connection.connect()
-        self.connection.send("<gmp/>")
-        self.connection.read()
-        self.connection.disconnect()
+        connection.connect()
+        connection.send("<gmp/>")
+        connection.read()
+        connection.disconnect()
```
Comment (bjoernricks) [inline tests/connections/test_unix_socket_connection.py:95]: It seems that all these already existing unix socket tests also don't ensure the behavior by asserting on something. I am really not sure if such tests are useful. IMHO they only test if no exception is raised.

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Not sure if the tests without assert statements should be changed. Personally I can't see an outcome for testing the expected behavior therefore I would love to change them.

Additionally checking the error messages should be done with assertRaises second parameter or with assertRaisesRegex for more complex messages.

**Comment by y0urself** [comment]
> Additionally checking the error messages should be done with assertRaises second parameter or with assertRaisesRegex for more complex messages.

Yeah. I started this as I found out, that it is possible. `assertRaises` should be extended everywhere in python-gvm and that means in **a lot** of places!

File: tests/connections/test_unix_socket_connection.py (hunk 2 above)
Comment (y0urself) [reply tests/connections/test_unix_socket_connection.py:53]: Maybe. But this again means: f-String everywhere (so better for another PR).

File: tests/connections/test_unix_socket_connection.py (hunk 3 above)
Comment (y0urself) [reply tests/connections/test_unix_socket_connection.py:101]: I have seen it, but was not sure about using it in a context_manager ...!

File: tests/connections/test_unix_socket_connection.py (hunk 4 above)
Comment (y0urself) [reply tests/connections/test_unix_socket_connection.py:95]: I will look at them again!

**Review by bjoernricks — CHANGES_REQUESTED** [review]

File: gvm/connections.py [hunk 5]
```diff
@@ -96,14 +95,14 @@ def __init__(self, timeout: Optional[int] = DEFAULT_TIMEOUT):
...
-    def _read(self):
+    def _read(self) -> bytes:
         return self._socket.recv(BUF_SIZE)
 
     def connect(self):
         """Establish a connection to a remote server"""
         raise NotImplementedError
 
-    def send(self, data: Union[bytes, str]):
+    def send(self, data: Union[bytes, str]) -> None:
```
[the remark this answers is not in the document]
Comment (y0urself) [reply gvm/connections.py:105]: according to https://docs.python.org/3/library/socket.html#socket.socket.sendall `sendall` returns `None` on success.

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**